frontend/src/Editor/LeftSidebar/SidebarPageSelector/PageHandler.jsx

```diff
@@ -126,36 +138,37 @@ export const PageHandler = ({
       <div className="card-body">
         <div className="row" role="button">
           <div className="col-auto">
-            {!isHovered && isHomePage && (
-              <img
-                className="animation-fade"
-                data-toggle="tooltip"
-                title="home page"
-                src="assets/images/icons/home.svg"
-                height={14}
-                width={14}
-                data-cy={'home-page-icon'}
-              />
+            {!isHovered && isHomePage && <Home width={16} height={16} />}
+            {/* When the page is hidden as well as disabled, disabled icon takes precedence */}
+            {!isHovered && (isDisabled || (isDisabled && isHidden)) && (
+              <FileRemove width={16} height={16} viewBox={'0 0 16 16'} />
+            )}
+            {!isHovered && isHidden && !isDisabled && <EyeDisable width={16} height={16} />}
+            {/* When hovered on disabled page, show disabled icon instead of hovered icon */}
+            {isHovered && isDisabled && <FileRemove width={16} height={16} viewBox={'0 0 16 16'} />}
+            {isHovered && !isDisabled && (
+              <div style={{ paddingRight: '4px' }}>
+                <SortableList.DragHandle show />
+              </div>
             )}
-            <SortableList.DragHandle show={isHovered} />
           </div>
           <div
-            className="col text-truncate font-weight-400 page-name"
+            className="col text-truncate font-weight-400 page-name tj-text-xsm  color-slate12"
             data-cy={`pages-name-${String(page.name).toLowerCase()}`}
+            style={isHomePage || isHidden || isHovered || isDisabled ? { paddingLeft: '0px' } : { paddingLeft: '16px' }}
           >
-            {page.name}
-          </div>
-          <div className="col-auto page-icons">
-            {isHidden && (
-              <img
-                data-toggle="tooltip"
-                title="hidden"
-                className="mx-2"
-                src="assets/images/icons/eye-off.svg"
-                height={14}
-                width={14}
-                data-cy={'hide-page-icon'}
-              />
+            {`${page.name}`}
+            {isIconApplied && (
+              <span
+                style={{
+                  marginLeft: '8px',
+                }}
+                className="color-slate09"
+              >
+                {isHomePage && 'Home'}
+                {(isDisabled || (isDisabled && isHidden)) && 'Disabled'}
```

Comment:
this condition is as good as simply checking for isDisabled right.?
`(isDisabled || (isDisabled && isHidden))` will be always equal to `isDisabled`

Reply:
Yes, removed isHidden.